server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java

```diff
@@ -681,6 +871,98 @@ public static List<Expression> removeWildcardInQueryFilterByDevice(
     }
   }
 
+  /**
+   * Concat measurement in Having clause with device path. And then, bind schema ({@link
+   * PartialPath} -> {@link MeasurementPath}) and removes wildcards.
+   *
+   * @return the expression list with full path and after binding schema
+   */
+  public static List<Expression> removeWildcardInHavingExpressionByDevice(
```

Comment:
This method is very similar to `removeWildcardInQueryFilterByDevice`, and we need to consider using one. For example, we should do some semantic checks upfront.